src/os/filestore/FileStore.cc

```diff
@@ -1873,10 +1898,17 @@ void FileStore::_do_op(OpSequencer *osr, ThreadPool::TPHandle &handle)
 
   osr->apply_lock.Lock();
   Op *o = osr->peek_queue();
-  apply_manager.op_apply_start(o->op);
+
+  if (!((g_conf->filestore_do_fast_sync) && (m_filestore_journal_writeahead))) {
```

Comment:
Tests like this don't make sense, filestore_do_fast_sync => m_filestore_journal_writeahead.  As I mentioned in another comment, please create a const m_fast_sync populated in the constructor based on the config values (and only set to true of the other config values allow it).